### ISSUE TITLE
[CON-238] Improved docs for "models" options in config.json

### ISSUE THREAD

**Issue by sestinj** [continuedev/continue#1344]
### Validations

- [X] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

There is a lack of documentation (outside of the reference page) for completionOptions, requestOptions, apiBase, apiKey, etc...

### Solution

It belongs on this page: https://docs.continue.dev/setup/configuration#self-hosting-an-open-source-model

and should mostly just in plain english cover

<sub>[CON-238](https://linear.app/continue/issue/CON-238/improved-docs-for-models-options-in-configjson)</sub>

**Comment by RomneyDa** [comment]
This has been added here: https://docs.continue.dev/reference
